uniffi_bindgen/src/bindings/ruby/templates/ErrorTemplate.rb

```diff
@@ -56,7 +56,7 @@ def to_s
 {%- for e in ci.enum_definitions() %}
 {% if ci.is_name_used_as_error(e.name()) %}
 {%- let typ=ci.get_type(e.name()).unwrap() %}
-{%- let canonical_type_name = canonical_name(typ.borrow()).borrow()|class_name_rb %}
+{%- let canonical_type_name = self::canonical_name(typ.borrow()).borrow()|class_name_rb %}
```

Comment:
huh, this will be the first use of `self::` in our templates - I didn't know that was a thing :) I'm sure we are doing quite a few things the hard way...

Reply:
There were some weird cases in askama, like how they handle methods. Now we're closer to rust in this regard, making it much simpler (imo) to do. Docs are [here](https://rinja.readthedocs.io/) but if anything is unclear or missing, please don't hesitate to tell me (or opening an issue on the project) so we can fix it.

Reply:
I think the tradeoff is requiring `self::` for static methods, but not requiring `self.` for regular methods which I will definitely take ([example](https://github.com/mozilla/uniffi-rs/blob/74c065a6ddb98320a1abe1155fd52db5677ca6ea/uniffi_bindgen/src/bindings/kotlin/templates/Types.kt#L136-L142))

Reply:
`self::` is to tell `rinja` that it's not a method actually. If you want a static method, it's either `Self::` or `TypeName::`. But in short, if it's a path (ie it contains `::`), `rinja` will not consider it a method.